### ISSUE TITLE
Can't call extension methods on `Object`

### ISSUE THREAD

**Issue by AnthonyDGreen** [dotnet/vblang#164]
This is a commonly reported pain-point. Because VB uses `Object` as its dynamic type we disallow extension method lookup on values typed as `Object`. This was done for back-compat concerns as a previously late-bound invocation could become an early-bound extension method invocation when taking a new version of a library, or importing a namespace, etc.

Some have suggested some kind of option which switches off `Object` as the dynamic type and that this would enable this scenario.

**Comment by franzalex** [comment]
I haven't tried this with `Object` types but I'm able to create extension methods that apply to instances of **all** types by leveraging generic type arguments as shown in the example below

``` vb
Public Sub DoSomething(Of T)(obj As T)
   ' code
End Sub
```

It works on *everything* I've thrown at it so far so to me, it's a non-issue.

**Comment by reduckted** [comment]
> I haven't tried this with Object types...

Yeah, unfortunately it doesn't work if the expression you want to call it on is `Object`. 

```vb
  Dim obj As Object = "some value"
  obj.DoSomething()
' ~~~~~~~~~~~~~~~ 
' ^ Option Strict On disallows late binding
```

_But_, as long as you're not dealing with an `Object` type, this is a great way of applying an extension method to all types. 👍
